mirdata/acousticbrainz_genre.py

```diff
@@ -0,0 +1,3800 @@
+
+Note, that the AllMusic ground-truth annotations are distributed separately at https://zenodo.org/record/2554044.
+
+
+
+Citation
+
+If you use the MediaEval AcousticBrainz Genre dataset or part of it, please cite our ISMIR 2019 overview paper:
+
+Bogdanov, D., Porter A., Schreiber H., Urbano J., & Oramas S. (2019).
+The AcousticBrainz Genre Dataset: Multi-Source, Multi-Level, Multi-Label, and Large-Scale.
+20th International Society for Music Information Retrieval Conference (ISMIR 2019).
+
+
+Acknowledgements
+
+This work is partially supported by the European Union’s Horizon 2020 research and innovation programme under grant agreement No 688382 AudioCommons.
+"""
+
+import csv
+import json
+
+import librosa
+import numpy as np
+import os
+
+from mirdata import download_utils
+from mirdata import jams_utils
+from mirdata import track
+from mirdata import utils
+
+DATASET_DIR = 'acousticbrainz_genre'
+REMOTES = {
+    'annotations': download_utils.RemoteFileMetadata(
+        filename='The AcousticBrainz Dataset Annotations.tar.gz',
+        url='http://isophonics.net/files/annotations/The%20AcousticBrainz Dataset%20Annotations.tar.gz',
+        checksum='62425c552d37c6bb655a78e4603828cc',
+        destination_dir='annotations',
+    )
+}
+
+DATA = utils.LargeData('acousticbrainz_genre_index.json')
+
+
+class Track(track.Track):
+    """AcousticBrainz Dataset track class
+
+    Args:
+        track_id (str): track id of the track
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+
+    Attributes:
+        audio_path (str): track audio path
+        beats_path (str): beat annotation path
+        chords_path (str): chord annotation path
+        keys_path (str): key annotation path
+        sections_path (str): sections annotation path
+        title (str): title of the track
+        track_id (str): track id
+
+    """
+
+    def __init__(self, track_id, data_home=None):
+        if track_id not in DATA.index:
+            raise ValueError('{} is not a valid track ID in AcousticBrainz genre Dataset'.format(track_id))
+
+        self.track_id = track_id
+
+        if data_home is None:
+            data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+        self._data_home = data_home
+        self._track_paths = DATA.index[track_id]
+        self.path = utils.none_path_join(
+            [self._data_home, self._track_paths['data'][0]]
+        )
+        data = self.track_id.split('#')
+        self.mbid = data[0]
+        self.mbid_group = data[1]
+
+    # Genre
+    @utils.cached_property
+    def genre(self):
+        """Genre: human-labeled genre and subgenres list"""
+        return [genre for genre in self.track_id.split('#')[2:]]
+
+    # Metadata
+    @utils.cached_property
+    def artist(self):
+        """Artist: metadata artist annotation
+        """
+        return load_extractor(self.path)["metadata"]["artist"]
```

Comment:
the only required data in acousticbrainz is the recording id. These items should use `.get()` to handle missing items.

Reply:
These cached properties load the file each time that you call a different one (the `load_extractor`). Maybe it would make more sense to cache the result of that method instead

Reply:
in this dataset, the value of `metadata.recording_id` is not always the same as the recording id used in the filename (what you call track_id here I think). Because of this, I think that there should also be a metadata property to load this metadata field.